app/domain/authentication/authn_k8s/initialize.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+require 'command_class'
+
+module Authentication
+  module AuthnK8s
+    # Secrets used when persisting a K8s authenticator
+    class K8sAuthenticatorData
+      include ActiveModel::Validations
+      include AuthData
+      attr_reader :service_account_token, :ca_certificate, :k8s_api_url, :json_data
+
+      def initialize(json_data)
+        @json_data = json_data
+        return if json_data.empty?
+        
+        @service_account_token = auth_parameters['service-account-token']
+        @ca_certificate = auth_parameters['ca-cert']
+        @k8s_api_url = auth_parameters['api-url']
+      end
+
+      def json_parameter_names
+        [ 'service-account-token', 'ca-cert', 'api-url' ]
+      end
+
+      def json_present?
+        !auth_parameters.keys.empty?
+      end
+
+      validates(
```

Comment:
What's the value in skipping validation if the JSON params are not present?

Reply:
Can we pass the environment variable into this context?

Reply:
We could always grab the value of the variable and use it, but that would also then force the user to have pre-configured the environment variable before calling the endpoint or it would error out. Is that what we want?

Reply:
Looks like this was a misread of the documentation on my part. The variables actually do need to be set in Conjur itself. I updated the class to reflect this.